packages/ra-core/src/core/useGetResourceLabel.ts

```diff
@@ -39,11 +39,7 @@ export const useGetResourceLabel = (): GetResourceLabel => {
                           smart_count: count,
                           _: resourceDefinition.options.label,
                       })
-                    : humanize(
-                          count > 1
-                              ? pluralize(resource)
-                              : singularize(resource)
-                      ),
+                    : humanize(inflect(resource, count)),
```

Comment:
Actually `inflection` is a tool to manage strings for the English language, so this fix works well for English, but it may break other languages (like French).
I'll need to make sure it's OK to change this with the team. Let's put this PR on hold for now. Thank you anyways!

Reply:
Consensus: applying English rules should be the default, using French gramary rules was a bug.
Users that use another language than English should provide translation keys for their resources.
Besides, this will only change the default value for resources that do not have a translation key, and when the smart count is 0, hence making this a pretty limited change.
TLDR: we're OK to merge!